### DISCUSSION TITLE
How to provide route-params to createServerData() called in routeData()

### DISCUSSION THREAD

**Discussion by benbender** [solidjs/solid-start#199] [Q&A] [answered]
Given the following code: 

```ts
export const routeData = ({ params }: RouteDataFuncArgs) => {
	console.log(params.projectId);

	return createServerData(async (projectId: string) =>
		db.project.findUniqueOrThrow({
			where: {
				id: projectId
			}
		})
	);
};
```

I'm struggeling to provide projectId to the `createServerData`-call as there seem to be no api to do it.
Given that `params` is available on the the client, `createServerData` executes on the server and `useRouteData` takes no args, I'm a bit lost here...
I could parse the `projectId` myself out of `request.url` but that doesn't seem to be the right way to go...

I couldn't find any doc/example on this usecase. Am I completely out of track or am I missing smth obvious?

**Comment by ryansolid** [comment, marked as answer]
It works the same way Resources work in Solid
```js
export const routeData = ({ params }: RouteDataFuncArgs) => {
	return createServerData(() => params.projectId, async (projectId: string) =>
		db.project.findUniqueOrThrow({
			where: {
				id: projectId
			}
		})
	);
};
```
This might seem weird at first but it lets us make a reactive connection so that if projectId ever changes we can trigger again. This can be done completely fine-grained per resource. We can also use this to make keys for invalidation. This first argument supports React Query-like keys.

**Reply by benbender** [reply]
Ah, thank you, got it! I never saw the source-parameter because of the overloading…
